tests/kernel/mellanox_config.pm

```diff
@@ -19,64 +21,44 @@ use testapi;
 use utils;
 use ipmi_backend_utils;
 use power_action_utils 'power_action';
-use version_utils 'is_sle';
-
-our $device1 = '/dev/mst/mt4119_pciconf0';
-our $device2 = '/dev/mst/mt4119_pciconf0.1';
-
-sub show_links {
-    assert_script_run("mlxconfig -d $device1 q|grep LINK_TYPE");
-    assert_script_run("mlxconfig -d $device2 q|grep LINK_TYPE");
-}
 
 sub run {
     my $self = shift;
-    $self->select_serial_terminal;
+    my $protocol = get_var('MLX_PROTOCOL', 2);
 
-    my $mft_version = get_required_var('MFT_VERSION');
-    my $protocol    = get_var('MLX_PROTOCOL') || 2;
+    # allow to configure SR-IOV and enable virtual functions
+    my $sriov_en = get_var('MLX_SRIOV',   0);
+    my $num_vfs  = get_var('MLX_NUM_VFS', 0);
```

Comment:
maybe we should skip MLX_SRIOV and enable it only if MLX_NUM_VFS variable is set. 

What happens if I give the job MLX_SRIOV=0 and MLX_NUM_VFS=32 ? It is assumed that if you want VFS, SRIOV must be enabled.

Reply:
Good point. I think we should default it to 0 if MLX_SRIOV is 0, to make sure the virtual functions get set to 0 again.